tests/api2/test_draid_record_and_block_size.py

```diff
@@ -0,0 +1,188 @@
+import pytest
+
+from middlewared.service_exception import ValidationErrors
+from middlewared.test.integration.assets.pool import another_pool
+from middlewared.test.integration.utils import call
+
+
+POOL_NAME = 'test_pool'
+
+
+@pytest.mark.parametrize(
+    'pool_type,record_size', [
+        ('DRAID1', '1M'),
+        ('MIRROR', '128K'),
+    ]
+)
+def test_pool_default_record_size(pool_type, record_size):
+    unused_disks = call('disk.get_unused')
+    if len(unused_disks) < 2:
+        pytest.skip('Insufficient number of disk to perform these test')
+
+    with another_pool({
+        'name': POOL_NAME,
+        'topology': {
+            'data': [{
+                'disks': [disk['name'] for disk in unused_disks[:2]],
+                'type': pool_type,
+                **({'draid_data_disks': 1} if pool_type.startswith('DRAID') else {})
+            }],
+        },
+        'allow_duplicate_serials': True,
+    }) as pool_name:
+        assert call('pool.dataset.get_instance', pool_name['name'])['recordsize']['value'] == record_size
+
+
+@pytest.mark.parametrize(
```

Comment:
We will have a relatively large number of pool creation/deletion operations here, running these tests might take up a very long time. Can we test the complete recordsize logic with an integration test and here only test a part of it?

Reply:
I'll use a fixture here so we avoid creating a pool each time these tests are run and that should solve this problem ✅

Reply:
Agree with @themylogin's review. When you use a fixture, make sure that the pool is cleaned up after this module is run. Thanks!